pkg/client/client_test.go

```diff
@@ -2304,3 +2307,59 @@ func assertErrorsMatch(t *testing.T, want, got []error) {
 		}
 	}
 }
+
+func TestPollUntil(t *testing.T) {
+	timeout := 150 * time.Millisecond
+	interval := 100 * time.Millisecond
+
+	// condition is met.
+	var lastErr1 error
+	err := pollUntil(context.Background(), timeout, interval, &lastErr1, func(ctx context.Context) (bool, error) {
```

Comment:
It is better that the tests verify Poll instead of helper?

Reply:
it'll not be possible to use it for all test cases as the timeout is large (5mins) which will reflect on the tests duration.

Using it only for some cases would be inconsistent  I think.

Reply:
See my comment above about having functional opts for Poll